textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/join/Join.java

```diff
@@ -22,16 +45,39 @@
 	 * @param inner is the inner operator producing the tuples
 	 * @param joinPredicate is the predicate over which the join is made
 	 */
-	public Join(IOperator outer, IOperator inner, IPredicate joinPredicate) {
-		this.outer = outer;
-		this.inner = inner;
-		this.joinPredicate = joinPredicate;
+	public Join(IOperator outerOperator, IOperator innerOperator, IPredicate joinPredicate) {
+		this.outerOperator = outerOperator;
+		this.innerOperator = innerOperator;
+		this.joinPredicate = (JoinPredicate) joinPredicate;
 	}
 
 	@Override
-	public void open() throws Exception {
-		// TODO Auto-generated method stub
-		
+	public void open() throws Exception, DataFlowException {
+		try {
+			outerOperator.open();
+			innerOperator.open();
+		} catch (Exception e) {
+			e.printStackTrace();
+			throw new DataFlowException(e.getMessage(), e);
+		}
+		if(!(joinPredicate.getjoinAttribute().getFieldType().equals(FieldType.STRING)||
+				joinPredicate.getjoinAttribute().getFieldType().equals(FieldType.TEXT))) {
+			outerOperator.close();
+			innerOperator.close();
+			throw new Exception("Fields other than \"STRING\" and \"TEXT\" are not supported by Join yet.");
+		}
+		// Load the inner tuple list into memory on open.
+		while((innerTuple = innerOperator.getNextTuple()) != null) {
+			innerTupleList.add(innerTuple);
+		}
+		// Close the inner operator as all the required tuples are already 
```

Comment:
I thought it would be best to just do it in the `getNextTuple()` method. Anyway the user checks for the `null` when calling the method. This makes it go well with existing code.